pkgs/development/compilers/rust/rustc.nix

```diff
@@ -227,6 +227,13 @@ in stdenv.mkDerivation (finalAttrs: {
 
     # Useful debugging parameter
     # export VERBOSE=1
+  '' + lib.optionalString stdenv.targetPlatform.isDarwin ''
+    # Replace hardcoded path to strip with llvm-strip
+    # https://github.com/NixOS/nixpkgs/issues/299606
+    substituteInPlace compiler/rustc_codegen_ssa/src/back/link.rs \
+      --replace-fail \
+        'let stripcmd = if cfg!(target_os = "macos") { "/usr/bin/strip" } else { "strip" };' \
+        'let stripcmd = "${lib.getExe' llvmPackages.llvm "llvm-strip"}";'
```

Comment:
This looks very fragile. Why not just replace /usr/bin/strip?

Reply:
Well, thanks to `replace-fail` it is not that fragile.
However, it might be unnecessary and we could only replace `/usr/bin/strip`.

Reply:
I made the change.